runelite-client/src/main/java/net/runelite/client/plugins/inventorygrid/BankGridOverlay.java

```diff
@@ -0,0 +1,239 @@
+	private final ItemManager itemManager;
+	private final InventoryGridPlugin plugin;
+
+	private static final ImmutableList<Varbits> TAB_VARBITS = ImmutableList.of(
+			Varbits.BANK_TAB_ONE_COUNT,
+			Varbits.BANK_TAB_TWO_COUNT,
+			Varbits.BANK_TAB_THREE_COUNT,
+			Varbits.BANK_TAB_FOUR_COUNT,
+			Varbits.BANK_TAB_FIVE_COUNT,
+			Varbits.BANK_TAB_SIX_COUNT,
+			Varbits.BANK_TAB_SEVEN_COUNT,
+			Varbits.BANK_TAB_EIGHT_COUNT,
+			Varbits.BANK_TAB_NINE_COUNT
+	);
+
+	@Inject
+	private BankGridOverlay(InventoryGridConfig config, Client client, ItemManager itemManager, InventoryGridPlugin plugin)
+	{
+		this.plugin = plugin;
+		this.itemManager = itemManager;
+		this.client = client;
+		this.config = config;
+
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		final Widget if3draggingWidget = client.getDraggedWidget();
+		final Widget bankWidget = plugin.getBankWidget();
+		final Widget bankInventoryWidget = plugin.getBankInventoryWidget();
+
+		final net.runelite.api.Point mouse = client.getMouseCanvasPosition();
+		final Point mousePoint = new Point(mouse.getX(), mouse.getY());
+
+
+		if (!client.isDraggingWidget() || client.getDraggedWidget() == null)
+		{
+			return null;
+		}
+
+		if (config.showBankGrid() && bankWidget != null && if3draggingWidget != null && if3draggingWidget.getParent() == bankWidget)
+		{
+
+			ItemContainer bankInventory = client.getItemContainer(InventoryID.BANK);
+
+			if (bankInventory == null)
+			{
+				return null;
+			}
+
+			int numItems = bankInventory.getItems().length;
+			if (numItems == 0)
+			{
+				return null;
+			}
+
+			int currentTab = client.getVar(Varbits.CURRENT_BANK_TAB);
+			int startIndex = 0;
+
+			if (currentTab > 0)
+			{
+				for (int i = currentTab - 1; i > 0; i--)
+				{
+					startIndex += client.getVar(TAB_VARBITS.get(i - 1));
+				}
+
+				int itemCount = client.getVar(TAB_VARBITS.get(currentTab - 1));
+				numItems = startIndex + itemCount;
+			}
+
+			for (; startIndex < numItems; ++startIndex)
+			{
+				Widget widget = bankWidget.getChild(startIndex);
+
+				Point bottomLeftPoint = new Point(widget.getCanvasLocation().getX(), widget.getCanvasLocation().getY() + widget.getHeight());
+				Point upperLeftPoint = new Point(widget.getCanvasLocation().getX(), widget.getCanvasLocation().getY());
+
+				final Rectangle bounds = widget.getBounds();
+				Rectangle intersection;
+
+				if (!bankWidget.getBounds().contains(bottomLeftPoint))
+				{
+					// set to the bottom of bank widget
+					intersection = bankWidget.getBounds().intersection(widget.getBounds());
+					bounds.setBounds(intersection);
+				}
+
+				if (!bankWidget.getBounds().contains(upperLeftPoint))
+				{
+					// set to top of bank widget
+					intersection = bankWidget.getBounds().intersection(widget.getBounds());
+					bounds.setBounds(intersection);
+				}
+
+				boolean inBounds = bounds.contains(mousePoint);
```

Comment:
This could use a better name.
```suggestion
				boolean mouseInBounds = bounds.contains(mousePoint);
```